src/Components/Locations.js

```diff
@@ -0,0 +1,49 @@
+import axios from "axios";
+import { useState } from "react";
+
+const Locations = () => {
+  const [locations, setLocations] = useState([]);
+  const [show, setShow] = useState("Show Locations");
+
+  const handleClick = async () => {
+    if (show === "Show Locations") {
+      try {
+        const res = await axios.get(
+          "https://ghibliapi.herokuapp.com/locations"
+        );
+        setLocations(res.data);
+        setShow("Hide Locations");
+      } catch (error) {
+        setLocations([]);
+      }
+    } else {
+        setLocations([]);
+        setShow("Show Locations")
+    }
+  };
+
+  return (
+    <section id="List of Locations">
+        <h1>List of Locations</h1>
+      <button onClick={handleClick}>{show}</button>
```

Comment:
Then if show is boolean you can use a ternary to decide what the button should say